sdk/go/common/apitype/ai.go

```diff
@@ -16,14 +16,46 @@ package apitype
 
 import "encoding/json"
 
-// Requests
+// SummarizeUpdateRequest
 
 type CopilotSummarizeUpdateRequest struct {
 	Query           string                 `json:"query"`
 	State           CopilotState           `json:"state"`
-	DirectSkillCall CopilotDirectSkillCall `json:"directSkillCall"`
+	DirectSkillCall CopilotSummarizeUpdate `json:"directSkillCall"`
 }
 
+type CopilotSummarizeUpdate struct {
+	Skill  string                       `json:"skill"` // The skill to call. e.g. "summarizeUpdate"
+	Params CopilotSummarizeUpdateParams `json:"params"`
+}
+
+type CopilotSummarizeUpdateParams struct {
+	PulumiUpdateOutput string `json:"pulumiUpdateOutput"` // The Pulumi update output to summarize.
+	Model              string `json:"model,omitempty"`    // The model to use for the summary. e.g. "gpt-4o-mini"
+	MaxLen             int    `json:"maxLen,omitempty"`   // The maximum length of the returned summary.
+}
+
+// SummarizePreviewRequest
+
+type CopilotSummarizePreviewRequest struct {
+	Query           string                  `json:"query"`
+	State           CopilotState            `json:"state"`
+	DirectSkillCall CopilotSummarizePreview `json:"directSkillCall"`
+}
+
+type CopilotSummarizePreview struct {
+	Skill  string                        `json:"skill"` // The skill to call. e.g. "summarizeUpdate"
```

Comment:
Huh, why would we allow calling `summarizeUpdate` same as above if we summarize a preview here?

Reply:
My bad! Copy paste